File: Recursion/Other/Easy/Factorial/Factorial.cpp
#include<bits/stdc++.h>
using namespace std;
long long factorial(int n, long long product)
{
    if (n <= 1)
    {
        return product;
    }
    return factorial(n - 1, product * n); // using tail recursion
}
int main(void) {
    cout << factorial(20, 1) << endl;
    return 0;
}

File: Recursion/Other/Easy/Factorial/Factorial.java
public class Factorial {
    static long factorial(int n, long product) {
        if(n <= 1) {
            return product;
        }
        return factorial(n-1, product*n); // using tail recursion
    }
    public static void main(String[] args) {
        System.out.println(factorial(20, 1));
    }
}